src/Main.hs

```diff
@@ -26,10 +26,8 @@ data Shape = Circle Vector Double
 
 -- Finds a,b, and c for a^2*x + b*x + c*x = 0, useful for finding intersections
 roots :: Double -> Double -> Double -> [Double]
-roots a b c = findRootsWithDescriminant (b * b - 4 * a *c) a b
-
-findRootsWithDescriminant :: Double -> Double -> Double -> [Double]
-findRootsWithDescriminant desc a b
-    | desc == 0 = [0.5 * (-b)]
-    | desc > 0 = [0.5 * (-b + sqrt desc), 0.5 * (-b - sqrt desc)]
+roots a b c
+    | descriminant == 0 = [0.5 * (-b)]
+    | descriminant > 0 = [0.5 * (-b + sqrt descriminant), 0.5 * (-b - sqrt descriminant)]
     | otherwise = []
+    where descriminant = b * b - 4 * a * c
```

Comment:
@rumblesan Would a better idea be to return Maybe [Double] instead?

This function only has three cases, so it's either a list with two elements, a list with one element, or an empty list. Maybe takes care of one possibility, but i'm not sure what the best option should be here for the return type.